### PR TITLE
`Development`: Add cypress run with postgres build plan

### PR DESCRIPTION
**Pull request by TheZoker**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

Our current setup for the cypress e2e tests only runs with a MySQL database. Since artemis with postgres tends to fail sometimes, we want to test that as well. 

### Description

This PR adds the config for a second cypress bamboo build plan. We use the already existing postgres docker compose file and adjust some envs, so artemis uses the correct database connection. We also needed to adjust the max connections, since the test did not run without it. 

#### Code Review
- [x] Review 1
- [x] Review 2

### CODE REVIEW

File: .bamboo/E2E-tests/execute-postgres.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+mv ./*.war build/libs/
+
+# pass current host's hostname to the docker container for server.url (see docker compose config file)
+export HOST_HOSTNAME=$(hostname)
+
+cd docker
+#just pull everything else than artemis-app as we build it later either way
+docker compose -f cypress-E2E-tests-postgres.yml pull artemis-cypress postgresql nginx
+docker compose -f cypress-E2E-tests-postgres.yml build --build-arg WAR_FILE_STAGE=external_builder --no-cache --pull artemis-app
+docker compose -f cypress-E2E-tests-postgres.yml up --exit-code-from artemis-cypress
```
Comment (4ludwig4) [inline .bamboo/E2E-tests/execute-postgres.sh:14]: Maybe it's better to add an argument for the docker compose .yml files and then add either ``cypress-E2E-tests-mysql.yml`` or ``cypress-E2E-tests-postgres.yml`` as argument in bamboo? But not 100 percent sure.

File: docker/cypress-E2E-tests-postgres.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                condition: service_started
+        volumes:
+            - ./nginx/artemis-nginx-cypress.conf:/etc/nginx/conf.d/artemis-nginx-cypress.conf:ro
+        ports:
+            - "80:80"
+            - "443:443"
+            # see comments in artemis/config/cypress-application-local.env why this port is necessary
+            - "54321:54321"
+
+    artemis-cypress:
```
Comment (4ludwig4) [inline docker/cypress-E2E-tests-postgres.yml:40]: I would create an ``artemis-cypress`` base service as for ``artemis-app`` this way you can leave out the command and the environment variables and just have to change them in one location.

**Review by 4ludwig4 — COMMENTED** [review]
Now also on GitHub and not just Slack ;)

Saw you already changed the Postgres command to exec form!

Besides that, I have two other "optimization" ideas.

File: docker/cypress-E2E-tests-postgres.yml (hunk 2 above)
Comment (4ludwig4) [reply docker/cypress-E2E-tests-postgres.yml:40]: Base service is looking good except that the ``depends_on:`` has to be inside of the children and can't be extended according to: https://docs.docker.com/compose/compose-file/#extends

File: docker/cypress-E2E-tests-postgres.yml [hunk 3]
```diff
@@ -0,0 +1,56 @@
+# ----------------------------------------------------------------------------------------------------------------------
+# Cypress Setup Postgres
+# ----------------------------------------------------------------------------------------------------------------------
+
+services:
+    postgres:
```
Comment (4ludwig4) [inline docker/cypress-E2E-tests-postgres.yml:6]:
```suggestion
    postgresql:
```

Would use the same service names as in the base services to cause less confusion when debugging.

**Review by 4ludwig4 — APPROVED** [review]
Looks good to me now after you integrated my feedback!
I hope the PostgreSQL problems with too many groups will also get resolved somehow.

**Review by Mtze — APPROVED** [review]

**Review by Mtze — APPROVED** [review]
Code changes are looking good to me!

**Pull request merged**